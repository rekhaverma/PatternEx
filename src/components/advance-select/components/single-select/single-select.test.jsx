import React from 'react';
import SingleSelect from './single-select.component';

/**
 * @status: Complete
 * @sign-off-by: Alex Andries
 */
describe('<SingleSelect />', () => {
  let props;

  beforeEach(() => {
    props = {
      options: [
        {
          id: 'string',
          value: 'string',
        }],
      activeOption: 'string',
      onOptionUpdate: jest.fn(),
      boxIsOpen: false,
      openBox: jest.fn(),
      closeBox: jest.fn(),
      allowClear: true,
    };
  });

  it('should match with snapshot', () => {
    const wrapper = shallow(<SingleSelect {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match with snapshot - with options list', () => {
    props.boxIsOpen = true;
    const wrapper = shallow(<SingleSelect {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
