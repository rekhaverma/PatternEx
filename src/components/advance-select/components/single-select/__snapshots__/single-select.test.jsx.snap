// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleSelect /> should match with snapshot - with options list 1`] = `
<div
  className="advance-select advance-select--single-select advance-select--open advance-select--clear-displayed"
>
  <div
    className="advance-select__selected-value"
    onClick={[Function]}
  >
    <span>
      string
    </span>
    <div
      className="advance-select__clear-button"
    >
      <Tooltip
        hideWhenHoveringTooltipContent={false}
        managerStyle={
          Object {
            "display": "flex",
          }
        }
        position="top"
        style={Object {}}
        trigger={
          <span
            className="icon-Trash-icon"
            onClick={[Function]}
          />
        }
      >
        <FormattedMessage
          id="dashboard.reset"
          values={Object {}}
        />
      </Tooltip>
    </div>
  </div>
  <OptionList
    allowSearch={false}
    className="advance-select"
    dropDownHeader={null}
    onOptionClick={[Function]}
    options={
      Array [
        Object {
          "id": "string",
          "isSelected": true,
          "value": "string",
        },
      ]
    }
  />
</div>
`;

exports[`<SingleSelect /> should match with snapshot 1`] = `
<div
  className="advance-select advance-select--single-select advance-select--clear-displayed"
>
  <div
    className="advance-select__selected-value"
    onClick={[Function]}
  >
    <span>
      string
    </span>
    <div
      className="advance-select__clear-button"
    >
      <Tooltip
        hideWhenHoveringTooltipContent={false}
        managerStyle={
          Object {
            "display": "flex",
          }
        }
        position="top"
        style={Object {}}
        trigger={
          <span
            className="icon-Trash-icon"
            onClick={[Function]}
          />
        }
      >
        <FormattedMessage
          id="dashboard.reset"
          values={Object {}}
        />
      </Tooltip>
    </div>
  </div>
</div>
`;
